bin/opds2_reaper_monitor

```diff
@@ -109,47 +110,61 @@ class OPDS2ReaperMonitor(OPDS2ImportMonitor):
             f"{self.missing_id_count} missing identifiers."
         )
 
+        # Number of ORM objects to buffer at a time.
+        query_batch_size = 500
+
         # Convert feed identifiers to our identifiers, so we can find them.
         # Unlike the import case, we don't want to create identifiers, if
         # they don't already exist.
+        self.log.info(
+            f"Mapping {len(self.seen_identifiers)} feed identifiers to database identifiers."
+        )
         identifiers, failures = Identifier.parse_urns(
             self._db, self.seen_identifiers, autocreate=False
         )
-        identifier_ids = [x.id for x in list(identifiers.values())]
+        identifier_count = len(identifiers)
+        failure_count = len(failures)
         if failures:
             self.log.warning(
-                f"Unable to parse {len(failures)} of {len(self.seen_identifiers)} identifiers."
+                f"Unable to parse {failure_count} of {identifier_count} identifiers."
             )
 
         collection_license_pools_qu = self._db.query(LicensePool).filter(
             LicensePool.collection_id == self.collection.id
         )
-        collection_license_pools = collection_license_pools_qu.count()
+        collection_count = collection_license_pools_qu.count()
 
-        unlimited_access_license_pools_qu = collection_license_pools_qu.filter(
+        eligible_license_pools_qu = collection_license_pools_qu.filter(
             LicensePool.licenses_available == LicensePool.UNLIMITED_ACCESS
         )
-        unlimited_access_license_pools = unlimited_access_license_pools_qu.count()
+        eligible_count = eligible_license_pools_qu.count()
 
-        # At this point we've gone through the feed and collected all the identifiers.
-        # If there's anything we didn't see, we know it's no longer available.
-        to_be_reaped_qu = unlimited_access_license_pools_qu.join(Identifier).filter(
-            ~Identifier.id.in_(identifier_ids)
-        )
```

Comment:
The key change starts here. We remove this join, which scaled poorly because the `SELECT` grew linearly with the number of identifiers in the feed.

The other part of this change is below, where we iterate over all eligible license pools locally and reap the ones that are not mentioned in the feed.